exonum/src/runtime/mod.rs

```diff
@@ -438,3 +441,60 @@ pub enum ApiChange {
     /// Instance has been removed.
     InstanceRemoved(InstanceId),
 }
+
+// TODO Write a full documentation when the interservice communications are fully implemented. [ECR-3493]
+/// Provide a low level context for the call of methods of the other service instance.
+#[derive(Debug)]
+pub struct CallContext<'a> {
+    /// Underlying execution context.
+    inner: &'a ExecutionContext<'a>,
+    /// Identifier of the caller service instance.
+    caller: InstanceId,
+    /// Identifier of the called service instance.
+    called: InstanceId,
+}
+
+impl<'a> CallContext<'a> {
+    /// Create a new call context for the specified execution context.
+    pub fn new(inner: &'a ExecutionContext<'a>, caller: InstanceId, called: InstanceId) -> Self {
+        Self {
+            inner,
+            caller,
+            called,
+        }
+    }
+
+    /// Perform the method call of the specified service interface.
+    pub fn call(
+        &self,
+        interface_name: impl AsRef<str>,
+        method_id: MethodId,
+        arguments: impl BinaryValue,
+        // TODO ExecutionError here mislead about the true cause of an occurred error. [ECR-3222]
+    ) -> Result<(), ExecutionError> {
+        let context = ExecutionContext {
+            fork: self.inner.fork,
+            dispatcher: self.inner.dispatcher,
+            actions: self.inner.actions.clone(),
+            caller: Caller::Service {
+                instance_id: self.caller,
+            },
+            call_stack_depth: self.inner.call_stack_depth + 1,
+        };
+        let call_info = CallInfo {
+            interface_name: interface_name.as_ref().to_owned(),
+            method_id,
+            instance_id: self.called,
+        };
+
+        assert!(
```

Comment:
Is it really has to be an assert? I think just returning an ExecutionError is enough, innit?